EIPS/eip-3155.md

```diff
@@ -0,0 +1,241 @@
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+This EIP introduces a new JSON standard for EVM traces. There has been a de-facto standard which is used by go-ethereum, OpenEthereum, Besu and Nethermind. This document shall document this standard as well as provide stricter guidelines to make new implementations easier.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+The Ethereum Virtual Machine executes all smart contract code on ethereum. In order to easier debug smart contracts and state tests, a de-facto standard was introduced to log every execution step of the EVM. This standard was implemented by go-ethereum, parity and nethermind and Besu. Since the de-facto standard was not well defined, the implementations differed slightly making it hard to develop adequate tooling.
+This EIP has multiple objectives:
+- Move the specification to a more visible place to encourage new clients to implement it
+- Strictly define corner cases that were not addressed in the previous version
+- Allow for updates to the specification in case new fields are introduced during execution
+- Provide sample output
+
+Implementing this EIP in all major clients allows us to create meaningful differential fuzzers that fuzz EVM implementations for the mainnet and all upcoming hardforks.
+It also helps finding differences in execution quickly in the case of a chain split.
+
+Due to the slight differences in output the following fields currently need to be dropped when comparing tracing outputs: $GasCost$, $MemorySize$, $RefundCounter$, $ReturnStack$, and $ReturnData$. This reduces the usefulness of tracing significantly. Creating a common standard would allow to add these fields thus increasing the surface that can be tested by a fuzzer.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Clients should be able to execute simple transactions as well as code and return traces. In the following, we will call this client CUT (client under test) and use go-ethereums `evm` binary for code examples. 
+
+### Inputs
+
+The CUT MUST be able to execute a statetest as described [here](https://www.ethdocs.org/en/latest/contracts-and-transactions/ethereum-tests/state_tests/index.html).
```

Comment:
We are blurring together the output of the trace with the way it is invoked.  I would prefer taking all references to the CUT out.  This format is also part of the JSON-RPC calls `debug_standardTraceBlockToFile` and `debug_standardTraceBadBlockToFile` which are not CLI clients and don't have standard error or standard out.  That would make this EIP just specifying the trace line format.  A separate EIP for using this for fuzzing with detail about the CUT could then be opened.